presto-docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,164 @@
+``X-Presto-Catalog``                   The catalog to be used when running the query.  Set by response header
+                                       ``X-Presto-Set-Catalog``.
+``X-Presto-Schema``                    The schema to be used when running the query.  Set by response header
+                                       ``X-Presto-Set-Schema``.
+``X-Presto-Time-Zone``                 The timezone to be used when running the query, which by default is the timezone of
+                                       the Presto engine.
+``X-Presto-Language``                  The language to be used when running the query and formatting results.  The language
+                                       of the session can be set on a per-query basis using the ``X-Presto-Language``
+                                       HTTP header, or via the ``PrestoConnection.setLocale(Locale)`` method in the
+                                       JDBC driver.
+``X-Presto-Trace-Token``               Supplies a trace token to the Presto engine to help identify log lines that originate
+                                       with this query request.
+``X-Presto-Session``                   Supplies a comma-separated list of name=value pairs as session properties.
+                                       When the Presto client run a ``SET SESSION name=value`` query, the name=value pair
+                                       is returned in the ``X-Set-Presto-Session`` response header, and added to the client's
+                                       list of session properties.
+                                       If the response header ``X-Presto-Clear-Session`` is returned, its value
+                                       is the name of a session property that will be removed from the client's accumulated
+                                       list.
+``X-Presto-Role``                      Sets the catalog role that will be used in this request.  Set by response header
+                                       ``X-Presto-Set-Role``.
+``X-Presto-Prepared-Statement``        A comma-separated list of the name=value pairs, where the names are names of
+                                       previously prepared SQL statements, and the values are ???.
+``X-Presto-Transaction-Id``            The transaction ID to be used when running the query.  Set by response header
+                                       ``X-Presto-Started-Transaction-Id`` and cleared by ``X-Presto-Clear-Transaction-Id``.
+``X-Presto-Client-Info``               Contains arbitrary information about the client program submitting the query.
+``X-Presto-Client-Tags``               A comma-separated list of "tag" strings, used to identify Presto resource groups.
+``X-Presto-Resource-Estimate``         A comma-separated list of ``resource=value`` assigments.  The possible choices
+                                       of ``resource`` are "EXECUTION_TIME", "CPU_TIME", with values specified as Java
```

Comment:
```suggestion
                                       of ``resource`` are "EXECUTION_TIME", "CPU_TIME", with values specified as airlift
```

I would also add some documentation on the format of these strings, since they're not inherent in the JDK.

Reply:
Changed to "airlift", and added text giving format documentation.